api/plugin_support.h

```diff
@@ -188,10 +254,32 @@ inst_set mambo_get_inst_type(mambo_context *ctx);
 int mambo_get_fragment_id(mambo_context *ctx);
 cc_type mambo_get_fragment_type(mambo_context *ctx);
 int mambo_get_inst_len(mambo_context *ctx);
+
+/**
+ * @brief Returns the address of the currently scanned untranslated instruction.
+ *
+ * Returns the untranslated address of the application's instructions that is
+ * currently being scanned by mambo using the plugin's context.
+ *
+ * @param ctx The context that describes the plugin.
+ * @return the untranslated instruction address.
+ */
 void *mambo_get_source_addr(mambo_context *ctx);
 int mambo_set_source_addr(mambo_context *ctx, void *source_addr);
 void *mambo_get_cc_addr(mambo_context *ctx);
 void mambo_set_cc_addr(mambo_context *ctx, void *addr);
+
+/**
+ * @brief Returns the id of the active thread.
```

Comment:
I wouldn't say active thread as many threads can execute at the same time. We should say something along the lines: ID of the thread where the callback was called?